### DISCUSSION TITLE
Porting from C#

### DISCUSSION THREAD

**Discussion by Imperatorn** [chances/teraflop-d#41] [Q&A] [answered]
Nice project. Saw that you ported from C#. Did you use any tools to help with that process?

I also want to convert som C# to D but I'm scared of async/await.

**Comment by chances** [comment, marked as answer]
Throughout the porting process I've referenced the the [Phobos Library docs](https://dlang.org/phobos/index.html) extensively. I have not used any automated tools for conversion and have been reimplementing features and refactoring them with the benefit of hindsight.

I don't miss `async`/`await` too much just yet. Though, future file I/O features of the engine would probably benefit from some more asynchrony. I'm implementing an event loop for the engine to support an [Actor model](https://github.com/chances/teraflop-d/blob/cc7ba063ef215ca5fab5684696833a11d886b522/source/teraflop/scripting/package.d#L34) for game components using [`libasync`](https://code.dlang.org/packages/libasync).

Thank you for your thoughtful question, @Imperatorn! 😄
